processing/src/main/java/io/druid/query/aggregation/cardinality/CardinalityAggregator.java

```diff
@@ -23,74 +23,50 @@
 import com.google.common.hash.Hasher;
 import com.google.common.hash.Hashing;
 import io.druid.query.aggregation.Aggregator;
+import io.druid.query.QueryDimensionInfo;
 import io.druid.query.aggregation.hyperloglog.HyperLogLogCollector;
-import io.druid.segment.DimensionSelector;
-import io.druid.segment.data.IndexedInts;
 
-import java.util.Arrays;
 import java.util.List;
 
 public class CardinalityAggregator implements Aggregator
 {
-  private static final String NULL_STRING = "\u0000";
-
-  private final List<DimensionSelector> selectorList;
+  private final String name;
+  private final List<QueryDimensionInfo<CardinalityAggregatorFactory.CardinalityAggregatorTypeHelper>> dimInfoList;
   private final boolean byRow;
 
-  private static final HashFunction hashFn = Hashing.murmur3_128();
-  public static final char SEPARATOR = '\u0001';
+  public static final HashFunction hashFn = Hashing.murmur3_128();
 
-  protected static void hashRow(List<DimensionSelector> selectorList, HyperLogLogCollector collector)
+  protected static void hashRow(List<QueryDimensionInfo<CardinalityAggregatorFactory.CardinalityAggregatorTypeHelper>> dimInfoList, HyperLogLogCollector collector)
```

Comment:
Maybe make `CardinalityAggregatorTypeHelper` top level type? Or rename it to just `TypeHelper`? This line is longer than 120 columns.